pkg/build/api/types.go

```diff
@@ -55,6 +55,9 @@ type BuildParameters struct {
 
 	// Output describes the Docker image the Strategy should produce.
 	Output BuildOutput `json:"output,omitempty"`
+
+	// Compute resource requirements to execute the build
+	Resources kapi.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
i'm talking about at the json level.  A user writing:

BuildConfig{
}

vs
BuildConfig{
  Resource: {}
}

In the first case we get nil and it's clear they made no effort to quota anything.

In the second case we get an empty slice and it's not clear what they were trying to do.

By making Resource not a pointer, you're turning the first case into the second case every time they don't specify anything.  It's no longer clear that they intentionally left it entirely blank, vs underspecifying the field.